pkgs/by-name/cy/cyrus-imapd/package.nix

```
@@ -0,0 +1,192 @@
+{
+  autoreconfHook,
+  bison,
+  brotli,
+  coreutils,
+  cunit,
+  cyrus-imapd,
+  cyrus_sasl,
+  fetchFromGitHub,
+  fig2dev,
+  flex,
+  icu,
+  jansson,
+  lib,
+  libbsd,
+  libcap,
+  libchardet,
+  libical,
+  libmysqlclient,
+  libsrs2,
+  libuuid,
+  libxml2,
+  makeWrapper,
+  nghttp2,
+  openssl,
+  pcre2,
+  perl,
+  pkg-config,
+  postgresql,
+  rsync,
+  shapelib,
+  sqlite,
+  stdenv,
+  testers,
+  unixtools,
+  valgrind,
+  withCalalarmd ? true,
+  withHttp ? true,
+  withJMAP ? true,
+  withMySQL ? false,
+  withNNTP ? false,
+  withPgSQL ? false,
+  withSQLite ? true,
+  withXapian ? true,
+  wslay,
+  xapian,
+  zlib,
+}:
+stdenv.mkDerivation rec {
+  pname = "cyrus-imapd";
+  version = "3.8.2";
+
+  src = fetchFromGitHub {
+    owner = "cyrusimap";
+    repo = "cyrus-imapd";
+    rev = "cyrus-imapd-${version}";
+    hash = "sha256-5I4EaLnxu2M1a79XmxUh1PXq703EfI73/9aohKg7jyc=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    pkg-config
+    autoreconfHook
+  ];
+  buildInputs =
+    [
+      unixtools.xxd
+      pcre2
+      flex
+      valgrind
+      fig2dev
+      perl
+      cyrus_sasl.dev
+      icu
+      jansson
+      libbsd
+      libuuid
+      openssl
+      zlib
+      bison
+      libsrs2
+    ]
+    ++ lib.optional stdenv.isLinux libcap
+    ++ lib.optionals (withHttp || withCalalarmd || withJMAP) [
+      brotli.dev
+      libical.dev
+      libxml2.dev
+      nghttp2.dev
+      shapelib
+    ]
+    ++ lib.optionals withJMAP [
+      libchardet
+      wslay
+    ]
+    ++ lib.optional withXapian [
+      rsync
+      xapian
+    ]
+    ++ lib.optional withMySQL libmysqlclient
+    ++ lib.optional withPgSQL postgresql
+    ++ lib.optional withSQLite sqlite;
+
+  enableParallelBuilding = true;
+
+  # Copied from debian deb package
+  patches = [
+    ./patches/0002-Shutdown-and-close-sockets-cleanly.patch
+    ./patches/0003-Fix-syslog-prefix.patch
+    ./patches/0009-Normalize-the-authentication-ID.patch
+    ./patches/0018-increase-test-timeout.patch
+    ./patches/0019-propagate-XXFLAGS.patch
+  ];
+
+  postPatch = ''
+    patchShebangs cunit/*.pl
+    patchShebangs imap/promdatagen
+    patchShebangs tools/*
+
+    echo ${version} > VERSION
+
+    substituteInPlace cunit/command.testc \
+      --replace-warn "/usr/bin/touch" "${coreutils}/bin/touch" \
+      --replace-warn "/bin/echo" "${coreutils}/bin/echo" \
+      --replace-warn "/usr/bin/tr" "${coreutils}/bin/tr" \
+      --replace-warn "/bin/sh" "${stdenv.shell}"
```

None of these quotes are necessary
```suggestion
      --replace-warn /usr/bin/touch ${coreutils}/bin/touch \
      --replace-warn /bin/echo ${coreutils}/bin/echo \
      --replace-warn /usr/bin/tr ${coreutils}/bin/tr \
      --replace-warn /bin/sh ${stdenv.shell}
```